File: math1/math1_1373_2_to_8.cpp
#if 0
# Ive attended the code.plus class (https://code.plus/)
# And I've referred it's codes
# And core parts can be come fully from the class.
#endif

#include <iostream>
#include <string>
using namespace std;
int main() {
    string s;
    cin >> s;
    int n = s.size();
    if (n%3 == 1) {
        cout << s[0];
    } else if (n%3 == 2) {
        cout << (s[0]-'0')*2 + (s[1]-'0');
    }
    for (int i=n%3; i<n; i+=3) {
        cout << (s[i]-'0')*4 + (s[i+1]-'0')*2 + (s[i+2]-'0');
    }
    cout << '\n';
    return 0;
}

File: math1/math1_1373_2_to_8.py
#if 0
# Ive attended the code.plus class (https://code.plus/)
# And I've referred it's codes
# And core parts can be come fully from the class.
#endif

s = input()
n = len(s)
ans = ''
if n%3 == 1:
    print(s[0], end='')
elif n%3 == 2:
    print((ord(s[0])-ord('0'))*2 + (ord(s[1])-ord('0')), end = '')
for i in range(n%3, n, 3):
    print((ord(s[i])-ord('0'))*4 + (ord(s[i+1])-ord('0'))*2 + (ord(s[i+2])-ord('0')), end = '')    